src/Features/LanguageServer/Protocol/Features/Diagnostics/DocumentAnalysisExecutor.cs

```diff
@@ -156,9 +156,14 @@ public async Task<IEnumerable<DiagnosticData>> ComputeDiagnosticsAsync(Diagnosti
             // Remap diagnostic locations, if required.
             diagnostics = await RemapDiagnosticLocationsIfRequiredAsync(textDocument, diagnostics, cancellationToken).ConfigureAwait(false);
 
-            if (span.HasValue)
+            if (span.HasValue && document != null)
             {
-                diagnostics = diagnostics.WhereAsArray(d => !d.HasTextSpan || span.Value.IntersectsWith(d.GetTextSpan()));
+                var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
teh call above to RemapDiagnosticLocationsIfRequiredAsync already required the contents of the document to process things.  so this shoudl be cheap.